### DISCUSSION TITLE
Booster and encrypt vs. sd-encrypt hook

### DISCUSSION THREAD

**Discussion by rieje** [anatol/booster#218] [General]
I've been reading about [kernel parameters from the Arch Wiki](https://wiki.archlinux.org/title/dm-crypt/System_configuration#Kernel_parameters).

Booster supports e.g. `rd.luks.uuid=`, does it mean it supports `sd-crypt` hooks only and not not `encrypt` hooks (e.g. for me, `cryptdevice=` that I had previous for mkinitcpio didn't work)? Or is encrypt vs. sd-encrypt hook only a mkinitcpio distinction and booster (and perhaps other initramfs generators) may support one, style, both, or maybe even their own?

Would it be accurate to say booster only supports kernel parameters read by `systemd-cryptsetup-generator`? I'm not sure how much of an assumption the Arch Wiki makes regarding mkinitcpio being the initramfs generator it talks about kernel parameters. I realize mkinitcpio is the default but it also seems kernel parameters is a standard that should be followed by initramfs generators.

**Comment by ghost** [comment]
@rieje Currently, [booster does not implement hooks as a feature](https://github.com/anatol/booster/issues/193) and it's not clear whether it will be. Also, because booster supports _some_ boot time kernel parameters from mkinitcpio or systemd-cryptsetup this does not imply hooks like sd-crypt will be supported
